wrapper/Tools/OpenMMMD.py

```diff
@@ -289,9 +289,41 @@ def writeSystemData(system, moves, Trajectory, block, softcore_lambda=False):
     moves_file.close()
 
 
-def centerSolute(system, space):
+def getSolute(system):
+    """Find the solute molecule based on the perturbed residue number.
+
+    Args:
+        system (system): The Sire system
+
+    Returns:
+        molecule: Molecule matching perturbed residue number assumed to be solvent
+    """
+
+    molecules = system[MGName("all")].molecules()
+    moleculeNumbers = molecules.molNums()
+    moleculeList = []
 
-    # ! Assuming first molecule in the system is the solute !
+    for moleculeNumber in moleculeNumbers:
+        molecule = molecules.molecule(moleculeNumber)[0].molecule()
+        moleculeList.append(molecule)
+
+    # Scan input to find a molecule with passed residue number
+
+    solute = None
+    for molecule in moleculeList:
+        if ( molecule.residue(ResIdx(0)).number() == ResNum(perturbed_resnum.val) ):
+            solute = molecule
+            break
+
+    if solute is None:
+        print ("FATAL ! Could not find a solute to perturb with residue number %s in the input !" 
+        "Check the value of your cfg keyword 'perturbed residue number'" % perturbed_resnum.val)
+        sys.exit(-1)
+
+    return solute
```

Comment:
Using the Python API to loop over molecules in a large system can be slow so it would be better to use the Sire search functionality, which will directly use C++ behind the scenes. For example, this could be re-written as:

```python
# Create the search query.
query = f"mol with resnum {perturbed_resnum.val}"

# Search the system for molecules containing a matching ResNum.
search = system.search(query)

# Make sure a single molecule matches the search.
# We don't want 0 molecules or duplicates. (Could handle errors for these separately if desired.)
if len(search) != 1:
    # Raise an exception rather than printing and using sys.exit(-1).
    raise Exception("FATAL! Could not find a solute to pertub...")

# Return the solute, i.e. the only search result.
return search[0]
```